### ISSUE TITLE
Consider adding prerelease tags and using these with puppet for ood.osc.edu

### ISSUE THREAD

**Issue by ericfranz** [OSC/ood-apps-installer#8]
@treydock wrote:

>The ood-app-installer update went through. Would it be possible to tag these changes? Right now web05 just stays in sync with master branch which has the potential to apply changes when we're not expecting.

@ericfranz wrote:

>We could tag them. However, we do not update the ood-apps-installer that often, and when we do are aware that we will be updating ood.os.edu. Currently we tag release versions. We could start tagging new versions of the ood-apps-installer using prerelease tags i.e. rc.1 rc.2. The negative is that it adds more steps when updating ood.osc.edu. If we could add a tagging scheme that didn't require us to open a new ticket every time we created a new pre-release or release tag that would be great.

Then @treydock wrote:
>
>Staying on the master branch is fine but it won't automatically update ood-app-installer since Puppet runs in "no apply" mode every 8 hours. This means Puppet reports changes and it takes someone from system's staff to actually run Puppet before those changes get applied.
>
>FWIW , if we went the tagged approach the same workflow could be used when WIAG wants to update the core packages on production OOD servers (mod_ood_proxy for example). The tag version is in Hiera (YAML) and changes could be applied using a merge request. Once a merge request is merged someone on system staff still has to sync the Puppet code to all masters and push out the change (no incident required if from merge request).
>
>One thing to watch out for is the branches in osc-puppetmaster-conf. They correspond to puppet environments and webtest* systems tend to be in test or test_ondemand environment and webdev* tend to be in dev environment. 
>
>The workflow still very rough as allowing additional groups to modify Puppet is still a new thing. I may need to sit down with WIAG at some point and work out git workflows as our use of branches is somewhat messy (requires rebasing somewhat frequently).

**Comment by ericfranz** [comment]
Original Service Now INCIDENT: INC0320377
